lasagne/layers/recurrent.py

```diff
@@ -99,6 +104,11 @@ def __init__(self, incoming, input_to_hidden, hidden_to_hidden,
         self.gradient_steps = gradient_steps
         self.grad_clipping = grad_clipping
         self.unroll_scan = unroll_scan
+        self.precompute_input = precompute_input
+
+        if unroll_scan and gradient_steps != -1:
+            raise RuntimeError(
```

Comment:
This should be a `ValueError`, not a `RuntimeError`; see https://docs.python.org/2/library/exceptions.html